src/test/groovy/ratpack/zipkin/internal/ZipkinHttpClientImplSpec.groovy

```diff
@@ -39,34 +46,136 @@ class ZipkinHttpClientImplSpec extends Specification {
+		where:
+			method | _
+			HttpMethod.GET | _
+			HttpMethod.POST | _
+			HttpMethod.PUT | _
+			HttpMethod.PATCH | _
+			HttpMethod.DELETE | _
+			HttpMethod.HEAD | _
+			HttpMethod.OPTIONS | _
 
-	def "get() should be traced"() {
+	}
+	def "Request returning 2xx include HTTP_URL annotation (but *not* status code)"(HttpResponseStatus status) {
+		given:
+			receivedResponse.getStatusCode() >> status.code()
+			requestSpec.getUri() >> uri
 		when:
-		ExecResult<ReceivedResponse> rs = harness.yield { e ->
-			zipkinHttpClient.get(uri, action)
-		}
+			harness.yield { e ->
+				zipkinHttpClient.get(uri, action)
+			}
+		then:
+			Span span = reporter.spans.get(0)
+		and: //should contain some binary annotations
+			assertThat(span.binaryAnnotations.findAll {it.key == TraceKeys.HTTP_STATUS_CODE}).isEmpty()
+			assertThat(span.binaryAnnotations.findAll {it.key == TraceKeys.HTTP_URL}).isNotEmpty()
+		where:
+			status | _
+			HttpResponseStatus.OK | _
+			HttpResponseStatus.ACCEPTED | _
+			HttpResponseStatus.ACCEPTED | _
+			HttpResponseStatus.NON_AUTHORITATIVE_INFORMATION | _
+			HttpResponseStatus.NO_CONTENT | _
+			HttpResponseStatus.RESET_CONTENT | _
+			HttpResponseStatus.PARTIAL_CONTENT | _
+	}
 
+	def "Request returning 4xx include HTTP_URL and HTTP_STATUS_CODE annotations"(HttpResponseStatus status) {
+		given:
+			receivedResponse.getStatusCode() >> status.code()
+			requestSpec.getUri() >> uri
+
+		when:
+			harness.yield { e ->
+				zipkinHttpClient.post(uri, action)
+			}
 		then:
-		rs.isSuccess()
-		rs.value == receivedResponse
-		reporter.spans.size() == 1
-		Span span = reporter.spans.get(0)
-		span.name == "get"
-		span.binaryAnnotations.findAll {it.key == TraceKeys.HTTP_STATUS_CODE}.size() == 0
-		span.binaryAnnotations.findAll {it.key == TraceKeys.HTTP_URL}.size() == 1
-		span.annotations.findAll {it.value == "cs"}.size() == 1
-		span.annotations.findAll {it.value == "cr"}.size() == 1
+			Span span = reporter.spans.get(0)
+		and: //should contain some binary annotations
+			assertThat(span.binaryAnnotations.findAll {it.key == TraceKeys.HTTP_STATUS_CODE}).isNotEmpty()
+			assertThat(span.binaryAnnotations.findAll {it.key == TraceKeys.HTTP_URL}).isNotEmpty()
+		where:
+			status | _
+			HttpResponseStatus.BAD_REQUEST | _
+			HttpResponseStatus.UNAUTHORIZED | _
+			HttpResponseStatus.PAYMENT_REQUIRED | _
+			HttpResponseStatus.FORBIDDEN | _
+			HttpResponseStatus.NOT_FOUND | _
+			HttpResponseStatus.METHOD_NOT_ALLOWED | _
+			HttpResponseStatus.NOT_ACCEPTABLE | _
+			HttpResponseStatus.PROXY_AUTHENTICATION_REQUIRED | _
+			HttpResponseStatus.REQUEST_TIMEOUT | _
+			HttpResponseStatus.CONFLICT | _
+			HttpResponseStatus.GONE | _
+	}
 
-		1 * httpClient.request(_, _) >> { URI u, Action<? super RequestSpec> a ->
-			assert u == uri
-			a.execute(requestSpec)
-			return Promise.value(receivedResponse)
-		}
-		1 * requestSpec.get() >> requestSpec
-		1 * receivedResponse.getStatusCode() >> 200
-		1 * requestSpec.getUri() >> uri
-		0 * _
```

Comment:
Good question.  I've always put them in the `then:` block though I guess they probably work the same.  Also I tend to put counts and `0 * _` to avoid surprises like calling an unexpected method or calling things multiple times on accident.   Guess its just a style preference.